src/hypothesis/extra/datetime.py

```diff
@@ -74,12 +101,19 @@ def do_draw(self, data):
 def datetimes(allow_naive=None, timezones=None, min_year=None, max_year=None):
     """Return a strategy for generating datetimes.
 
+    .. deprecated:: 3.9.0
+        use :py:func:`hypothesis.strategies.datetimes` instead.
+
     allow_naive=True will cause the values to sometimes be naive.
     timezones is the set of permissible timezones. If set to an empty
-    collection all timezones must be naive. If set to None all available
-    timezones will be used.
+    collection all datetimes will be naive. If set to None all timezones
+    available via pytz will be used.
+
+    All generated datetimes will be between min_year and max_year, inclusive.
 
     """
+    note_deprecation('This strategy was deprecated in version 3.9.0 - use '
+                     'the improved version in hypothesis.strategies')
```

Comment:
Looking further down – are we expecting to use this deprecation warning for all the strategies in this module? If so, ick. 😞

Deprecation warnings should be specific and instructive in what to do next, and a generic message can't do that.

(I would offer more specific advice, but I don't have a working interpreter to experiment with right now.)